src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig

```diff
@@ -165,31 +165,26 @@
                         tbody.appendChild(rows);
 
                         if (infoSpan) {
-                            var text = requestStack.length + ' call' + (requestStack.length > 1 ? 's' : '');
+                            var text = requestStack.length + ' AJAX request' + (requestStack.length > 1 ? 's' : '');
                             infoSpan.textContent = text;
                         }
                     } else {
-                        var cell = document.createElement('td');
-                        cell.setAttribute('colspan', '4');
-                        cell.textContent = "No AJAX requests yet.";
-                        var row = document.createElement('tr');
-                        row.appendChild(cell);
-                        tbody.appendChild(row);
+                        var ajaxToolbarPanel = document.querySelector('.sf-toolbar-block-ajax');
+                        ajaxToolbarPanel.style.display = 'none';
                     }
                 }
 
                 requestCounter[0].textContent = requestStack.length;
 
-                var className = 'sf-toolbar-ajax-requests sf-toolbar-status';
-                if (state == 'ok') {
-                    className += ' sf-toolbar-status-green';
-                } else if (state == 'error') {
-                    className += ' sf-toolbar-status-red';
+                var className = 'sf-toolbar-ajax-requests sf-toolbar-value';
+                requestCounter[0].className = className;
+
+                var ajaxToolbarPanel = document.querySelector('.sf-toolbar-block-ajax');
```

Comment:
you should query for the toolbar panel only once. Here, you do it on line 172 and here.

and the DOM query should be done at the beginning of the method, with other DOM queries, to avoid invalidating the DOM several times